python/graphstorm/gconstruct/remap_result.py

```diff
@@ -318,32 +463,77 @@ def main(args, gs_config_args):
     assert out_chunk_size > 0, \
         f"Output chunk size should larger than 0 but get {out_chunk_size}."
 
+    # TODO remap embeddings
+
+    ################## remap prediction #############
     # if pred_etypes (edges with prediction results)
-    # is not None, We need to remap edge prediction results.
+    # is not empty, we need to remap edge prediction results.
     # Note: For distributed SageMaker runs, pred_etypes must be
-    # provided if edge prediction result rempa is required,
-    # as result_info.json is only saved by rank0 and there is no shared fs.
-    if pred_etypes is not None:
+    # provided if edge prediction result remap is required,
+    # as result_info.json is only saved by rank0 and
+    # there is no shared file system.
+    if len(pred_etypes) > 0:
         for pred_etype in pred_etypes:
             assert os.path.exists(os.path.join(predict_dir, "_".join(pred_etype))), \
                 f"prediction results of {pred_etype} do not exists."
-    elif os.path.exists(os.path.join(predict_dir, "result_info.json")):
-        # User does not provide pred_etypes.
-        # Try to get it from saved prediction config.
-        with open(os.path.join(predict_dir, "result_info.json"),
-                    "r",  encoding='utf-8') as f:
-            info = json.load(f)
-            pred_etypes = [etype.split(",") for etype in info["etypes"]] \
-                    if "etypes" in info else []
-
-    ntypes = \
-        [etype[0] for etype in pred_etypes] + \
+
+    # If pred_ntypes (nodes with prediction results)
+    # is not empty, we need to remap edge prediction results
+    # Note: For distributed SageMaker runs, pred_ntypes must be
+    # provided if node prediction result remap is required,
+    # as result_info.json is only saved by rank0 and
+    # there is no shared file system.
+    if len(pred_ntypes) > 0:
+        for pred_ntype in pred_ntypes:
+            assert os.path.exists(os.path.join(predict_dir, pred_ntype)), \
+                f"prediction results of {pred_ntype} do not exists."
+
+    if with_shared_fs:
+        # Only when shared file system is avaliable,
+        # we will check result_info.json for
+        # pred_etypes and pred_ntypes.
+        # If shared file system is not avaliable
+        # result_info.json is not guaranteed to exist
+        # on each instances. So users must use
+        # --pred-ntypes or --pred-etypes instead.
+        #
+        # In case when both --pred-ntypes or --pred-etypes
+        # are provided while result_info.json is also avaliable,
+        # GraphStorm remaping will follow --pred-ntypes or --pred-etypes
+        # and ignore the result_info.json.
+        if os.path.exists(os.path.join(predict_dir, "result_info.json")):
```

Comment:
what does result_info.json provide?

Reply:
https://github.com/awslabs/graphstorm/blob/id-remap-dev/python/graphstorm/model/utils.py#L1099-L1103
the result_info.json provides: which node type(s) or edge type(s) has the prediction results saved and also how many partitions are there.

Reply:
i feel that we have multiple files that try to save the metadata. do users need to understand these files? i'm a little worried that there will be too many such files and eventually get out of control.

Reply:
Most users do not need to understand these configs if they are just using the cmdline to run graphstorm.